docs/dpg-archboard.md

```diff
@@ -0,0 +1,166 @@
+as a landing place for documentation. But when we add them, they will be non-breaking because you can still access them as a dict.
+
+```python
+response = client.get_object()
+assert response["hello"] == "world"
+```
+
+### Creating a POST request
+
+POST requests are another area where we're going to see JSON bodies pop up.
+
+We've also additionally added overloads for post methods where the input is a JSON type, for example a model. These overloads will be helpful to people
+who don't want to read large models into memory just to pass them as a JSON input. Instead, they are now able to stream serialized JSON straight to the
+service.
+
+```python
+
+client.put_object({"hello": "world"})
+
+with open("myLargeJsonFile", "rb") as fd:
+    client.put_object(fd)
+```
+
+### Break the glass scenario
+
+Our DPG clients all come with a `send_request` function on a client. Here, you can create your own request
+and use our client to send it to the service. This way you get all of the existing pipeline setup for free,
+and can send your request
+
+```python
+from azure.core.rest import HttpRequest
+from azure.purview.catalog import PurviewCatalogClient
+from azure.identity import DefaultAzureCredential
+
+client = PurviewCatalogClient(endpoint="endpoint", credential=DefaultAzureCredential())
+
+request = HttpRequest("GET", "http://new-endpoint.com")
+response = client.send_request(request)
+response.raise_for_status()
+json_response = response.json()
+```
+
+--------------------------------------------------------------------------
+--------------------------------- Changlong ---------------------------------
+
+### Streams
+
+#### Inputs
+
+As mentioned above, we've opened up streamed inputs to include more cases, so users with large input bodies aren't forced
+to read their bodies into memory before passing them to the service. Otherwise, streamed inputs are largely the same
+as they are in our current generation
+
+```python
+with open("my_file", "rb") as fd:
+    client.stream_file(fd)
+```
+
+#### Outputs
+
+With streamed outputs, users used to iterate over `.stream_download()` on the response object. We've changed this method to
+`iter_bytes()`. This is the syntax that `httpx` uses, which is the HTTP stack that Python is migrating to.
+More aligned with `httpx`.
+
+- Ask about what other languages are doing for retry streams
+
+
+### Multiple Media Types
+
+Here is a place where we lied about not being completely break proof, and that is because of the added functionality
+of our overloads.
+
+Say a service team starts out by just accepting a JSON input to an endpoint. Then our initial generation of this SDK
+will also allow users to pass a streamed body with default content type "application/json". If the service team
+then adds a streamed body that they accept and add content type `"application/octet-stream"`, then the default content type
+for streamed input bodies will change from `"application/json"` to `"application/octet-stream"`.
```

Comment:
Hi @iscai-msft , I don't figure out why the default type changes from `json` to `octet-stream`. What's the rule in codegen to decide which content type should be the default one?

Reply:
oh, find out should be here?https://github.com/Azure/autorest.python/blob/archboard_docs/autorest/m4reformatter/__init__.py#L388.
Seems json will be prior to the octet-stream